### ISSUE TITLE
Fix S1144 FP: Public indexer in nested class used in parent class

### ISSUE THREAD

**Issue by fiotti** [SonarSource/sonar-dotnet#9652]
### Description

Public indexer in a private inner class is erroneously always considered unused.

### Repro steps

```csharp
class Test
{
    class Inner
    {
        public int this[int test] => 0; // S1144: Remove the unused property 'this[]'.
    }

    public Test()
    {
        Inner inner = new();
        _ = inner[0]; // <-- used here
    }
}
```

### Expected behavior

I didn't expect any diagnostics to be emitted.

### Actual behavior

False positive.

### Known workarounds

```csharp
#pragma warning disable S1144
```

### Related information

* SonarLint for Visual Studio 2022, version 8.9.0.92083
* Visual Studio Professional 2022 (64 bit), version 17.10.4
* dotnet 8.0.303
* MSBuild version 17.10.4+10fbfbf2e for .NET Framework, version 17.10.4.21802
* Windows 11 Pro 23H2 22631.3880

**Comment by sebastien-marichal** [comment]
Hello @fiotti,

I confirm this is a false positive!

Thank you for reporting it :)
